plugins/outputs/timestream/timestream.go

```diff
@@ -233,11 +241,21 @@ func (t *Timestream) writeToTimestream(writeRecordsInput *timestreamwrite.WriteR
 				return t.createTableAndRetry(writeRecordsInput)
 			}
 			t.logWriteToTimestreamError(notFound, writeRecordsInput.TableName)
+			// log error and return error to telegraf to retry in next flush interval
+			// We need this is to avoid data drop when there are no tables present in the database
+			return fmt.Errorf("Failed to write to Timestream database '%s' table '%s'. Skipping metric! Error: '%s'",
+				t.DatabaseName, *writeRecordsInput.TableName, err)
```

Comment:
Errors should start with lowercase letters according to convention. Furthermore, sending the metric is retried in the next flush cycle, so ` Skipping metric!` is misleading I think.

Reply:
Updated, thank you